src/common/tests/checksum_timing.c

```diff
@@ -0,0 +1,353 @@
+/**
+ * (C) Copyright 2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+#include <stdint.h>
+#include <inttypes.h>
+
+#include <getopt.h>
+#include <daos/checksum.h>
+#include <gurt/common.h>
+
+#ifdef MCHECKSUM_SUPPORT
+#include <mchecksum.h>
+#include <mchecksum_error.h>
+#endif
+
+static int
+timebox(int (*cb)(void *), void *arg, uint64_t *usec)
+{
+	struct timespec start, end, result;
+
+	clock_gettime(CLOCK_MONOTONIC_RAW, &start);
+
+	int rc = cb(arg);
+
+	clock_gettime(CLOCK_MONOTONIC_RAW, &end);
+
+	result = d_timediff(start, end);
+
+	uint64_t delta_us = ((uint64_t)result.tv_sec * 1000000000 +
+				(uint64_t)result.tv_nsec) / 1000;
```

Comment:
sorry I didn't notice but this patch still have a few C99 declare cases. But I'm OK to give a pass if you can fix them in a separate patch.